devito/ir/equations/equation.py

```diff
@@ -110,32 +112,27 @@ def __new__(cls, *args, **kwargs):
                              "and kwargs=%s" % (str(args), str(kwargs)))
 
         # Well-defined dimension ordering
-        ordering = dimension_sort(expr, key=lambda i: not i.is_Time)
-
-        # Introduce space sub-dimensions if need to
```

Comment:
note for reviewers: moved to operator.py because the substitution has to be applied globally to all equations, rather than locally. Otherwise, we might create redundant yet symbolically identical SubDimensions